modules/authentication/src/handlers/phone.ts

```diff
@@ -0,0 +1,174 @@
+import ConduitGrpcSdk, {
+  ConduitRouteActions,
+  ConduitRouteReturnDefinition,
+  ConduitString,
+  GrpcError,
+  ParsedRouterRequest,
+  RoutingManager,
+  SMS,
+  UnparsedRouterResponse,
+} from '@conduitplatform/grpc-sdk';
+import { ConfigController } from '../config/Config.controller';
+import { isEmpty, isNil } from 'lodash';
+import { status } from '@grpc/grpc-js';
+import { AccessToken, RefreshToken, Token, User } from '../models';
+import { AuthUtils } from '../utils/auth';
+import { TokenType } from '../constants/TokenType';
+import { ISignTokenOptions } from '../interfaces/ISignTokenOptions';
+import moment = require('moment');
+
+export class PhoneHandlers {
+  private sms: SMS;
+  private initialized: boolean = false;
+
+  constructor(private readonly grpcSdk: ConduitGrpcSdk, private readonly routingManager: RoutingManager) {
+
+  }
+
+  async validate(): Promise<Boolean> {
+    const config = ConfigController.getInstance().config;
+    let errorMessage = null;
+    await this.grpcSdk.config
+      .moduleExists('sms')
+      .catch((e: any) => (errorMessage = e.message));
+    if (config.phoneAuthentication.enabled && !errorMessage) {
+      // maybe check if verify is enabled in sms module
+      await this.grpcSdk.waitForExistence('sms');
+      this.sms = this.grpcSdk.sms!;
+      this.initialized = true;
+      return true;
+    } else {
+      console.log('sms phone authentication not active');
+      return false;
+    }
+  }
+
+  async declareRoutes() {
+    this.routingManager.route(
+      {
+        path: '/phone',
+        action: ConduitRouteActions.POST,
+        description: `Endpoint that can be used to authenticate with phone. 
+              A message will be returned which indicates that a verification code has been sent.`,
+        bodyParams: {
+          phone: ConduitString.Required,
+        },
+      },
+      new ConduitRouteReturnDefinition('LoginResponse', {
+        message: ConduitString.Required,
+      }),
+      this.authenticate.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/phone-login',
```

Comment:
make it something like /phone/verify, since both endpoints work both for login and register. So maybe the first one should be /phone and the second one /phone/verify or /phone/authenticate